presto-main/src/main/java/com/facebook/presto/operator/DriverContext.java

```diff
@@ -159,24 +159,23 @@ public void recordBlocked(ListenableFuture<?> blocked)
 
     public void finished()
     {
+        freeMemory(memoryReservation.get());
+        freeSystemMemory(systemMemoryReservation.get());
```

Comment:
@cberner shouldn't all the memory be freed up by closing `OperatorContexts` first? Freeing memory in driver seems like an ugly workaround for some problem somewhere else. 

I know that this may not be a part of this PR and I don't want to block this fixup (since imo it's pretty urgent), but maybe we should have some followup work after this?